### ISSUE TITLE
Joystick scanner error

### ISSUE THREAD

**Issue by ghostsquad** [hybridgroup/gobot#502]
```
$ go run ./platforms/joystick/bin/scanner.go
# command-line-arguments
platforms/joystick/bin/scanner.go:45:50: cannot use t.Which (type sdl.JoystickID) as type int in argument to sdl.JoystickOpen
```

Not sure what to do about this error. I have a 360 controller plugged in, and verified that it's detected and working through this:
https://github.com/360Controller/360Controller

**Comment by deadprogram** [comment]
Looks like either a bug or an API change. Either way, needs to be looked into.

**Comment by deadprogram** [comment]
Hi @ghostsquad I just committed this branch with what should address this issue: https://github.com/hybridgroup/gobot/tree/bugfix/joystick-scanner

If you have a chance, can you please give it a try? I am currently travelling and do not have any joystick/gamepad to try out.

**Comment by ghostsquad** [comment]
I'll try it out. Stay tuned.

**Comment by deadprogram** [comment]
I've added code and merged into the `dev` branch, which is working on my machine:

```
$ ./scanner 
Joystick 0 connected
[5880 ms] Axis: 4       value:-32768
[5880 ms] Axis: 5       value:-32768
```

I will leave this issue open until the next release.

**Comment by ghostsquad** [comment]
Sorry I haven't had a chance to check this out. I'm terrible.

**Comment by gherlein** [comment]
this branch fixes the error for me, thanks.

**Comment by deadprogram** [comment]
This has now been released as part of v1.10.0 so now closing this issue. Thank you everyone!
